src/lex.js

```diff
@@ -1390,7 +1407,7 @@ Lexer.prototype = {
     // Check regular expression for correctness.
 
     try {
-      new RegExp(body, flags.join(""));
+      new RegExp(body);
```

Comment:
Thanks, @mathiasbynens ! Cait was right to ask for your input :) I plan to incorporate that logic from Esprima into this patch, but I had an idea for a small improvement. I just opened https://github.com/jquery/esprima/pull/1180 to propose it.